test/faktory_worker/job/job_integration_test.exs

```diff
@@ -0,0 +1,40 @@
+defmodule FaktoryWorker.JobIntegrationTest do
+  use ExUnit.Case
+
+  alias FaktoryWorker.Job
+
+  describe "perform_async/3" do
+    test "should send a new job to faktory" do
+      faktory_name = :"Test_#{random_string()}"
+      start_supervised!({FaktoryWorker, name: faktory_name, pool: [size: 1]})
+
+      queue_name = random_string()
```

Comment:
Was random string needed to avoid conflicts because it's not being cleared up afterwards?

Reply:
Correct, this in theory allows us to run this test in async mode when we have more of them and also provides a very unlikely chance of having a naming collision. Once we can clear up queues (`ACK`ing the jobs) we won't need to do this for the queue name anymore.